src/pages/Calendar/Calendar.tsx

```diff
@@ -1,16 +1,59 @@
 import "./Calendar.scss";
 import Layout from "../../components/Layout/Layout";
-import calendarImg from "../../assets/images/calendar.svg";
 import Header from "../../components/Header/Header";
 import { useState } from "react";
+import "react-modern-calendar-datepicker/lib/DatePicker.css";
+import { Calendar } from "@hassanmojab/react-modern-calendar-datepicker";
+import { format } from "date-fns";
+import EventCard from "../../components/EventCard/EventCard";
+import { Event } from "../../types/types";
 
-const Calendar = () => {
+type CalendarPageProps = {
+  eventData: Event[];
+};
+
+const CalendarPage = ({ eventData }: CalendarPageProps) => {
   const [isActive, setIsActive] = useState(true);
+  const [buttonVariants, setButtonVariants] = useState<boolean[]>(
+    new Array(eventData.length).fill(false)
+  );
 
   const handleClick = () => {
     setIsActive(!isActive);
   };
 
+  const handleClickButton = (eventIndex: number) => {
+    const newButtonVariants = [...buttonVariants];
+    newButtonVariants[eventIndex] = !newButtonVariants[eventIndex];
+    setButtonVariants(newButtonVariants);
+  };
+
+  const currentDate = new Date();
+  const defaultValue = {
+    year: currentDate.getFullYear(),
+    month: currentDate.getMonth() + 1,
+    day: currentDate.getDate(),
+  };
+  // eslint-disable-next-line
```

Comment:
I do not think you need to disable this line anymore?

Reply:
The tests kept failing before this so adding this was Ollie's suggestion and it was the only resolution.

Reply:
yeah keep it in, the build fails as the linter doesn't like the use of any but it's the only way I could resolve it

Reply:
Interesting, I will run the tests and see

Reply:
Oh I see you are using any we want to remove that.

```tsx
// ADD IN THE DayValue TYPE FROM THE PACKAGE YOU ARE USING
import {
  Calendar,
  DayValue,
} from "@hassanmojab/react-modern-calendar-datepicker";

// THE DEFAULT VALUE IS THIS TYPE NOW
  const defaultValue: DayValue = {
    year: currentDate.getFullYear(),
    month: currentDate.getMonth() + 1,
    day: currentDate.getDate(),
  };
// THE STATE IS THIS TYPE NOW
  const [selectedDay, setSelectedDay] = useState<DayValue>(defaultValue);
// THE DayValue TYPE CAN NE NULL OR UNDEFINED WE CAN ADD A FALLBACK OPTION JUST IN CASE
  const timeStampDay = new Date(
    `${selectedDay?.year || defaultValue.year}, ${
      selectedDay?.month || defaultValue.month
    }, ${selectedDay?.day || defaultValue.day}`
  );
```

Reply:
told you he would be able to solve it